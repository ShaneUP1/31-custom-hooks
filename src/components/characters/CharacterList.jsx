import React from 'react';
import PropTypes from 'prop-types';
import SingleCharacter from './SingleCharacter';
import { Link } from 'react-router-dom';

const CharacterList = ({ characters }) => {
  const CharacterElements = characters.map(character => {
    if(character.image){
      return (
        <li key={character.id}>
          <Link to={`/characters/${character.id}`} >
            <SingleCharacter name={character.name} image={character.image} />
          </Link>
        </li>
      );
    }
    return;
  });
   
  return (
    <ul data-testid="characters">
      {CharacterElements}
    </ul>
  );
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string
  })).isRequired
};

export default CharacterList;
