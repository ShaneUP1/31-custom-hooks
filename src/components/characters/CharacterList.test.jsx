/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders a list of characters', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList characters={[{ name: 'Ari', image: 'www.url.com', id: '2' }]}/>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
