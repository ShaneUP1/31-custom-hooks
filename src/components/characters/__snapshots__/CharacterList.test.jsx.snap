// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders a list of characters 1`] = `
<DocumentFragment>
  <ul
    class="list"
    data-testid="characters"
  >
    <li>
      <a
        href="/characters/2"
      >
        <figure>
          <img
            alt="Ari"
            src="www.url.com"
          />
          <figcaption>
            Ari
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
